src/pages/NFTs/OpenBidNFT/BidModal.tsx

```diff
@@ -181,20 +197,76 @@ const MESSAGE = styled.div`
   }
 `
 
-export const BidModal: FC<{ setVisible: (x: boolean) => void; visible: boolean }> = (props) => {
-  const { setVisible, visible } = props
+export const BidModal: FC<{ setVisible: (x: boolean) => void; visible: boolean; details?: ISingleNFT }> = (props) => {
+  const { setVisible, visible, details } = props
   const [mode, setMode] = useState('bid')
+  const { sessionUser, fetchSessionUser } = useNFTProfile()
+  const { connected, publicKey } = useWallet()
+  const { bidOnSingleNFT } = useNFTCollections()
   const [isLoading, setIsLoading] = useState(false)
+  const [bidPrice, setBidPrice] = useState('000.000')
 
   const onCancel = () => setMode('bid')
   const reviewBid = () => setMode('review')
-  const confirmBid = () => {
-    setIsLoading(true)
-    // TODO: Fake API
-    setTimeout(() => {
+
+  useEffect(() => {
+    if (connected && publicKey) {
+      if (!sessionUser || sessionUser.pubkey !== `${publicKey}`) {
+        fetchUser()
+      }
       setIsLoading(false)
-      setVisible(false)
-      setMode('bid')
+    } else {
+      setIsLoading(false)
+      notify({
+        type: 'error',
+        message: (
+          <MESSAGE>
+            <div>Couldn't fetch user data, please connect your wallet and refresh this page.</div>
+          </MESSAGE>
+        )
+      })
+    }
+
+    return () => {}
+  }, [publicKey, connected])
+
+  const fetchUser = () => {
+    fetchSessionUser('address', `${publicKey}`).then((res) => {
+      if (!res || (res.response && res.response.status !== 200) || res.isAxiosError) {
+        notify({
+          type: 'error',
+          message: (
+            <MESSAGE>
+              <div>Couldn't fetch user data, please refresh this page.</div>
+            </MESSAGE>
+          )
+        })
+      }
+    })
+  }
+
+  const confirmBid = async () => {
+    setIsLoading(true)
+    const bidObject = {
+      clock: Date.now() + '', // string, not a number
+      tx_sig: 'RANDOM_TX_SIG_HERE',
```

Comment:
The data needed for the bid post request is not complete but you can send a bid and receive a true response but the data doesn't update which I;m guessing the reason is the tokens and keys are not filled yet